src/test/java/org/fogbowcloud/manager/requests/api/local/http/AttachmentOrdersControllerTest.java

```diff
@@ -0,0 +1,395 @@
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import junit.framework.Assert;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(SpringRunner.class)
+@WebMvcTest(value = AttachmentOrdersController.class, secure = false)
+@PrepareForTest(ApplicationFacade.class)
+public class AttachmentOrdersControllerTest {
+    private final String ATTACHMENT_ENDPOINT =
+            "/".concat(AttachmentOrdersController.ATTACHMENT_ENDPOINT);
+
+    private final String CORRECT_BODY =
+            		"{"
+                    + "\"source\": \"b8852ff6-ce00-45aa-898d-ddaffb5c6173\","
+                    + "\"target\": \"596f93c7-06a1-4621-8c9d-5330a089eafe\","
+                    + "\"device\": \"/dev/sdd\""
+                    + "}";
+
+    private final String BODY_WITH_EMPTY_PROPERTIES =
+            		"{"
+                    + "\"source\": \"b8852ff6-ce00-45aa-898d-ddaffb5c6173\","
+                    + "\"target\": \"596f93c7-06a1-4621-8c9d-5330a089eafe\","
+                    + "\"device\": \"/dev/sdd\""
+                    + "}";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private ApplicationFacade facade;
+
+    @Before
+    public void setUp() throws FogbowManagerException {
+        this.facade = Mockito.spy(ApplicationFacade.class);
+        PowerMockito.mockStatic(ApplicationFacade.class);
+        given(ApplicationFacade.getInstance()).willReturn(this.facade);
+    }
+
+    // test case: Request a attachment creation and test successfully return. 
+    // Check the response of request and the call of facade for create the attachment.
+    @Test
+    public void CreateAttachmenTest() throws Exception {
+    	
+    	// set up
+        String orderId = "fake-id";
+        Mockito.doReturn(orderId)
+                .when(this.facade)
+                .createAttachment(Mockito.any(AttachmentOrder.class), Mockito.anyString());
+        
+        RequestBuilder requestBuilder =
+                createRequestBuilder(
+                        HttpMethod.POST, ATTACHMENT_ENDPOINT, getHttpHeaders(), CORRECT_BODY);
+
+        // exercise: Make the request
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        int expectedStatus = HttpStatus.CREATED.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        Assert.assertEquals(orderId, result.getResponse().getContentAsString());
+        
+        Mockito.verify(this.facade, Mockito.times(1))
+        	.createAttachment(Mockito.any(AttachmentOrder.class), Mockito.anyString());
+    }
+
+    
+    // test case: Request a attachment creation with 3 types of wrong body and test fail return
+    // Check the response of request and the call of facade for create the attachment.
```

Comment:
@GustavoDinizMonteiro nice test case. I'd suggest to divide it in three independent cases, though.

Reply:
i think it's a good idea, i was trying to get reuse of the configuration code, but I see that the test would be clearer if divided into 3.